### PR TITLE
Add ecommerce tracking documentation

### PR DESCRIPTION
**Pull request by JamesCGDS**
## What
This PR adds documentation for the `ga4-ecommerce-tracker.js` script.

## Why
To keep the code for the GA4 implementation well documented.

## Visual Changes
N/A

### CODE REVIEW

File: docs/analytics-ga4/ga4-ecommerce-tracker.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+//= require ./analytics-ga4/ga4-ecommerce-tracker
+```
+
+In `live_search.js` in the `finder-frontend` repository (https://github.com/alphagov/finder-frontend/blob/main/app/assets/javascripts/live_search.js#L129):
+
+```JavaScript
+GOVUK.analyticsGa4.Ga4EcommerceTracker.init(isNewPage) // isNewPage is a boolean and is used to detect whether a new page has been loaded or not
+```
+
+Unlike the other GA4 tracking scripts in this repository, ecommerce tracking is not initialised by `init-ga4.js`. Instead, it is initialised by the JS that runs on finder pages (specifically, the `GOVUK.Ga4EcommerceTracking()` function). It is called:
```
Comment (AshGDS) [inline docs/analytics-ga4/ga4-ecommerce-tracker.md:23]: Is `GOVUK.Ga4EcommerceTracking()` correct?

File: docs/analytics-ga4/ga4-ecommerce-tracker.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+                    "item_list_name": "Search"
+                },
+                "etc..."
+            ]
+        }
+    }
+}
+```
+
+For further information on the object structure/schema, please see the implementation guide: https://docs.google.com/document/d/1LGLRb6zMvJBjzMYPgLAcBWbS-GsKGUzLcumCsZgiUhM
```
Comment (AshGDS) [inline docs/analytics-ga4/ga4-ecommerce-tracker.md:71]: This isn't a public document, so I don't think it should be linked here

**Review by AshGDS — CHANGES_REQUESTED** [review]
Looks good, though I have requested a couple changes :+1:

File: docs/analytics-ga4/ga4-ecommerce-tracker.md (hunk 2 above)
Comment (JamesCGDS) [reply docs/analytics-ga4/ga4-ecommerce-tracker.md]: Good point, have removed. Thank you!

File: docs/analytics-ga4/ga4-ecommerce-tracker.md (hunk 1 above)
Comment (JamesCGDS) [reply docs/analytics-ga4/ga4-ecommerce-tracker.md]: It isn't, it should say `LiveSearch.prototype.Ga4EcommerceTracking()` as you rightly pointed out

**Pull request merged**